component/client-handler/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidatorTest.java

```diff
@@ -217,43 +216,62 @@ public Object[][] createJWT() throws Exception {
+                        "not-expired is not failed", null, false},
+                {jsonWebToken6, properties2, true, "Valid JWT token with custom issuer validation should pass.", null, false},
                 {jsonWebToken7, properties3, false, "JWT persisted in database with preventTokenReuse " +
-                        "enabled is not failed."},
+                        "enabled is not failed.", null, false},
                 {jsonWebToken9, properties1, false, "JWT persisted in database with preventTokenReuse " +
-                        "disabled is not failed."},
+                        "disabled is not failed.", null, false},
                 {jsonWebToken10, properties4, true, "Valid JWT token with custom audience validation should pass" +
-                        "."},
-                {jsonWebToken11, properties1, false, ""},
-                {jsonWebToken12, properties5, true, ""},
-                {jsonWebToken12, properties5, true, ""},
-                {jsonWebToken13, properties1, false, ""},
-                {jsonWebToken15, properties1, false, ""},
-                {jsonWebToken16, properties4, false, ""},
-                {jsonWebToken17, properties6, false, ""},
-                {jsonWebToken18, properties7, false, ""},
-                {jsonWebToken19, properties1, true, "Unable to use same JTI across tenants."},
+                        ".", null, false},
+                {jsonWebToken11, properties1, false, "", null, false},
+                {jsonWebToken12, properties5, true, "", null, false},
+                {jsonWebToken12, properties5, true, "", null, false},
+                {jsonWebToken13, properties1, false, "", null, false},
+                {jsonWebToken15, properties1, false, "", null, false},
+                {jsonWebToken16, properties4, false, "", null, false},
+                {jsonWebToken17, properties6, false, "", null, false},
+                {jsonWebToken18, properties7, false, "", null, false},
+                {jsonWebToken19, properties1, true, "Unable to use same JTI across tenants.", null, false},
                 {jsonWebToken20, properties1, false, "Duplicated JTI was used in same tenant with " +
-                        "preventTokenReuse enabled."},
-                {jsonWebToken21, properties1, true, "JWT with valid audience from the accepted value list should pass."}
+                        "preventTokenReuse enabled.", null, false},
+                {jsonWebToken23, properties1, true, "JWT with valid audience from the accepted value list should pass.",
+                        null, false},
+                {jsonWebToken22, properties1, true, "JWT with registered signing algorithm should pass.", "RS256", false},
+                {jsonWebToken22, properties1, false, "JWT with unregistered signing algorithm should fail.", "PS256", false},
+                // Assuming that RS512 is a FAPI allowed signing algorithm
```

Comment:
Lets mention the reason for the assumption here.

Reply:
Fixed in 96f2fe65da4c330ca560a495e94ad95d0db88298